src/components/dex/tokenSelector.tsx

```diff
@@ -70,17 +73,24 @@ export default function TokenSelector({
 			return;
 		}
 
-		onChange(e, 0);
+		onChange(e, '0');
 	};
 
 	const handleAmountChange = (e: ChangeEvent<HTMLInputElement>) => {
-		if (Number(e.target.value) > 10000000) {
+		let input = e.target.value;
+		const dpLen = count_decimals(input)
+		if(dpLen > MAX_DP) {
+			input = fixDP(input)
+		}
+		const num = Number(input);
+		if(Number.isNaN(num) || num < 0 || num > 10000000) {
```

Comment:
this check "isNan || num < 0 || num > 100000000" appears often, maybe make a helper func?

Reply:
I have made a checkNumInput() to those code.